Zend/zend_compile.c

```diff
@@ -7463,9 +7656,82 @@ void zend_compile_class_decl(znode *result, zend_ast *ast, zend_bool toplevel) /
 			opline->result.opline_num = -1;
 		}
 	}
+
+	return ce;
 }
 /* }}} */
 
+static void zend_declare_enum_case_constant(zend_ast *ast, zend_class_entry *enum_class)
+{
+	zend_string *enum_case_name = zend_ast_get_str(ast->child[0]);
+	zend_string *enum_class_name = enum_class->name;
+
+	zval class_name_zval;
+	ZVAL_STR_COPY(&class_name_zval, enum_class_name);
+	zend_ast *class_name_ast = zend_ast_create_zval_ex(&class_name_zval, ZEND_NAME_NOT_FQ);
+
+	zval case_name_zval;
+	ZVAL_STR_COPY(&case_name_zval, enum_case_name);
+	zend_ast *case_name_ast = zend_ast_create_zval_ex(&case_name_zval, ZEND_NAME_NOT_FQ);
+
+	zend_ast *case_value_zval_ast = NULL;
+	zend_ast *case_value_ast = ast->child[1];
+	if (case_value_ast != NULL) {
+		znode case_value_node;
+		zend_compile_expr(&case_value_node, case_value_ast);
+		if (case_value_node.op_type != IS_CONST) {
+			zend_error_noreturn(E_ERROR, "Enum case value must be constant");
```

Comment:
What's the reason for this over zend_error_noreturn(E_COMPILE_ERROR?

Reply:
It's copied code, `E_COMPILE_ERROR` is indeed more fitting :slightly_smiling_face: